corehq/apps/domain/views.py

```diff
@@ -532,27 +532,17 @@ def test_repeater(request, domain):
     url = request.POST["url"]
     repeater_type = request.POST['repeater_type']
     format = request.POST.get('format', None)
+    repeater_class = repeater_types[repeater_type]
     form = GenericRepeaterForm(
         {"url": url, "format": format},
         domain=domain,
-        repeater_class=repeater_types[repeater_type]
+        repeater_class=repeater_class
     )
     if form.is_valid():
         url = form.cleaned_data["url"]
-        # now we fake a post
-        def _stub(repeater_type):
-            if 'case' in repeater_type.lower():
-                return CaseBlock(
-                    case_id='test-case-%s' % uuid.uuid4().hex,
-                    create=True,
-                    case_type='test',
-                    case_name='test case',
-                ).as_string()
-            else:
-                return "<?xml version='1.0' ?><data id='test'><TestString>Test post from CommCareHQ on %s</TestString></data>" % \
-                       (datetime.datetime.utcnow())
-
-        fake_post = _stub(repeater_type)
+        format = format or RegisterGenerator.default_format_by_repeater(repeater_class)
+        generator_class = RegisterGenerator.generator_class_by_repeater_format(repeater_class, format)
+        fake_post = generator_class(repeater_class()).get_test_payload()
```

Comment:
actually it does look like some of the payloads are json. we may want to pass the format into `simple_post`: https://github.com/dimagi/dimagi-utils/blob/master/dimagi/utils/post.py#L45